### ISSUE TITLE
[need assistance] ingamesn/devname blacklist

### ISSUE THREAD

**Issue by kyle95wm** [barronwaffles/dwc_network_server_emulator#183]
Okay so basically I've had a few Mii names come onto my server that are deemed "inappropriate" and thought "this should be easy to blacklist names instead of disabling the player's profile"

Well Im here to say I'm having a hell of a time getting it to work

https://github.com/kyle95wm/dwc_network_server_emulator/commit/f8d9092974100c9d6de1ab6a0dab44b44457b754

For SOME reason I'm getting this:

[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:49242] Received availability request for 'mariokartwii' from 192.168.101.197:49242...
[2015-07-06 00:25:37 | NasServer] Request to /ac from ('192.168.101.197', 36066)
[2015-07-06 00:25:37 | NasServer] {'lang': '01', 'gamecd': 'KMCE', 'makercd': '01', 'unitcd': '1', 'ipaddr': '192.168.101.197', 'region': '01', 'userid': '7394979570907', 'gsbrcd': 'RMCJ2j5uevc', 'cfc': '3640650471079990', 'ingamesn': '\x00n\x00o\x00 \x00n\x00a\x00m\x00e', 'devtime': '150706002523', 'csnum': 'LU1780337239', 'action': 'login', 'macadr': '0017abbd3401', 'sdkver': '001000'}
127.0.0.1 - - [06/Jul/2015 00:25:37] "POST /ac HTTP/1.1" 200 -
[2015-07-06 00:25:37 | GamespyDatabase] [yXJQ6MMz] STARTING: SELECT COUNT(_) FROM ip_banned WHERE ipaddr = 192.168.101.197
[2015-07-06 00:25:37 | GamespyDatabase] [yXJQ6MMz] DONE: Took 0.00390005111694 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [WFiqfQwI] STARTING: SELECT COUNT(_) FROM console_macadr_banned WHERE macadr = 0017abbd3401
[2015-07-06 00:25:37 | GamespyDatabase] [WFiqfQwI] DONE: Took 0.000633001327515 real time / 0.01 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [QlUGp9h8] STARTING: SELECT COUNT(_) FROM console_cfc_banned WHERE cfc = 3640650471079990
[2015-07-06 00:25:37 | GamespyDatabase] [QlUGp9h8] DONE: Took 0.000624895095825 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [L6hKHxGL] STARTING: SELECT COUNT(_) FROM console_csnum_banned WHERE csnum = LU1780337239
[2015-07-06 00:25:37 | GamespyDatabase] [L6hKHxGL] DONE: Took 0.000596046447754 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [4q3hxNiR] STARTING: SELECT COUNT(_) FROM ingamesn_banned WHERE ingamesn = no name
[2015-07-06 00:25:37 | GamespyDatabase] [4q3hxNiR] DONE: Took 0.000606060028076 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [k6GhHmww] STARTING: SELECT COUNT(_) FROM nas_logins WHERE authtoken = NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ
[2015-07-06 00:25:37 | GamespyDatabase] [k6GhHmww] DONE: Took 0.000660181045532 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [xpmjNSov] STARTING: SELECT \* FROM nas_logins WHERE userid = 7394979570907
[2015-07-06 00:25:37 | GamespyDatabase] [xpmjNSov] DONE: Took 0.000775098800659 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [JQrRlCW4] STARTING: UPDATE nas_logins SET authtoken = NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ, data = {"lang": "01", "gamecd": "KMCE", "makercd": "01", "unitcd": "1", "challenge": "MBCF0y9s", "ipaddr": "192.168.101.197", "region": "01", "userid": "7394979570907", "gsbrcd": "RMCJ2j5uevc", "cfc": "3640650471079990", "ingamesn": "AG4AbwAgAG4AYQBtAGU_", "devtime": "150706002523", "csnum": "LU1780337239", "action": "login", "macadr": "0017abbd3401", "sdkver": "001000"} WHERE userid = 7394979570907
[2015-07-06 00:25:37 | GamespyDatabase] [JQrRlCW4] DONE: Took 0.00145101547241 real time / 0.0 processor time
[2015-07-06 00:25:37 | NasServer] login response to ('192.168.101.197', 36066)
[2015-07-06 00:25:37 | NasServer] {'locator': 'gamespy.com', 'token': 'NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ', 'retry': '0', 'returncd': '001', 'challenge': 'MBCF0y9s', 'datetime': '20150706002537'}
[2015-07-06 00:25:37 | GameSpyProfileServer] [192.168.101.197:61964] Received connection from 192.168.101.197:61964
[2015-07-06 00:25:37 | GameSpyProfileServer] [192.168.101.197:61964] SENDING: '\lc\1\challenge\ZQANZYWRTW\id\1\final\'...
[2015-07-06 00:25:37 | GameSpyProfileServer] [192.168.101.197:61964] RESPONSE: '\login\challenge\caXKkpb09olDMeI5T8AZQTaj410oY36U\authtoken\NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ\partnerid\11\response\b0d8e2a66866072aedc239867e8f8508\firewall\1\port\0\productid\11059\gamename\mariokartwii\namespaceid\16\sdkrevision\3\quiet\0\id\1\final\'...
[2015-07-06 00:25:37 | GamespyDatabase] [mfQTlOap] STARTING: SELECT COUNT(_) FROM ip_banned WHERE ipaddr = 192.168.101.197
[2015-07-06 00:25:37 | GamespyDatabase] [mfQTlOap] DONE: Took 0.00246906280518 real time / 0.0 processor time
[2015-07-06 00:25:37 | GameSpyProfileServer] [192.168.101.197:61964] {'authtoken': 'NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ', 'gamename': 'mariokartwii', 'firewall': '1', 'namespaceid': '16', 'challenge': 'caXKkpb09olDMeI5T8AZQTaj410oY36U', '**cmd_val**': '', 'quiet': '0', 'port': '0', 'sdkrevision': '3', '**cmd**': 'login', 'partnerid': '11', 'id': '1', 'login': '', 'response': 'b0d8e2a66866072aedc239867e8f8508', 'productid': '11059'}
[2015-07-06 00:25:37 | GamespyDatabase] [NIdIW8JF] STARTING: SELECT data FROM nas_logins WHERE authtoken = NDSCn4Gwu0RaZVGOjO6amcv55MEnmkbtZrI1VxQWhodH9WDUBdo89xWUFdUuFPDMY6bmoED9YieQGsmApRZ
[2015-07-06 00:25:37 | GamespyDatabase] [NIdIW8JF] DONE: Took 0.000490188598633 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [CbSZJ4Qj] STARTING: SELECT COUNT(_) FROM users WHERE userid = 7394979570907 AND gsbrcd = RMCJ2j5uevc
[2015-07-06 00:25:37 | GamespyDatabase] [CbSZJ4Qj] DONE: Took 0.00051212310791 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [ROuasH8l] STARTING: SELECT \* FROM users WHERE userid = 7394979570907 and gsbrcd = RMCJ2j5uevc
[2015-07-06 00:25:37 | GamespyDatabase] [ROuasH8l] DONE: Took 0.000909090042114 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [Qzmstvog] STARTING: SELECT COUNT(_) FROM users WHERE profileid = 4
[2015-07-06 00:25:37 | GamespyDatabase] [Qzmstvog] DONE: Took 0.000437021255493 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [rYWVBbzT] STARTING: DELETE FROM sessions WHERE profileid = 4
[2015-07-06 00:25:37 | GamespyDatabase] [rYWVBbzT] DONE: Took 0.00102019309998 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [8vV34FfT] STARTING: SELECT COUNT(_) FROM sessions WHERE session = 94919902
[2015-07-06 00:25:37 | GamespyDatabase] [8vV34FfT] DONE: Took 0.000466823577881 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [S7vKuql8] STARTING: INSERT INTO sessions VALUES (94919902, 4, cjVpakI3cU9xQXA4bHFyVg__)
[2015-07-06 00:25:37 | GamespyDatabase] [S7vKuql8] DONE: Took 0.00092601776123 real time / 0.01 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [qyemBVYh] STARTING: SELECT \* FROM buddies WHERE userProfileId = 0 AND blocked = 0
[2015-07-06 00:25:37 | GamespyDatabase] [qyemBVYh] DONE: Took 0.00070595741272 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [5QmVleZL] STARTING: SELECT \* FROM buddies WHERE userProfileId = 0 AND blocked = 1
[2015-07-06 00:25:37 | GamespyDatabase] [5QmVleZL] DONE: Took 0.000531911849976 real time / 0.0 processor time
[2015-07-06 00:25:37 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] SENDING: \lc\2\sesskey\94919902\proof\8c4a719b1940521ccfe223e78e27b9db\userid\7394979570907\profileid\4\uniquenick\6n73i2p6rRMCJ2j5uevc\lt\cjVpakI3cU9xQXA4bHFyVg__\id\1\final\
[2015-07-06 00:25:37 | GamespyDatabase] [8eX51Ak5] STARTING: SELECT \* FROM pending_messages WHERE targetid = 4
[2015-07-06 00:25:37 | GamespyDatabase] [8eX51Ak5] DONE: Took 0.000500917434692 real time / 0.01 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [TxG8CyBx] STARTING: SELECT \* FROM buddies WHERE userProfileId = 4 AND blocked = 0
[2015-07-06 00:25:37 | GamespyDatabase] [TxG8CyBx] DONE: Took 0.000214099884033 real time / 0.0 processor time
[2015-07-06 00:25:37 | GamespyDatabase] [8vz189hR] STARTING: SELECT \* FROM buddies WHERE userProfileId = 4 AND blocked = 0
[2015-07-06 00:25:37 | GamespyDatabase] [8vz189hR] DONE: Took 0.000205039978027 real time / 0.0 processor time
[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Received heartbeat from 192.168.101.197:61798... localip010.0.1.30localport61798natneg1statechanged1gamenamemariokartwiipublicip0publicport0numplayers0maxplayers0unknown4unknown0unknown90unknown1unknown0unknown0unknown0
[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:61798 129bafce] publicport 0 doesn't match localport 61798, so changing publicport to 61798...
[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Sent challenge to 192.168.101.197:61798...
[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Received challenge from 192.168.101.197:61798... lQRTODRQLBGr7GdIAutlWr0UMwYA
[2015-07-06 00:25:37 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Sent client registered to 192.168.101.197:61798...
[2015-07-06 00:25:37 | GamespyBackendServer] Deleted 0 mariokartwii servers where session = 312192974
[2015-07-06 00:25:37 | GamespyBackendServer] Added {'natneg': '1', '**session**': 312192974, 'localport': '61798', 'unknown': '0', 'gamename': 'mariokartwii', 'numplayers': '0', 'maxplayers': '0', 'publicip': '-983193408', '**console**': 0, 'localip0': '10.0.1.30', 'publicport': '61798', 'statechanged': '1'} to the server list for mariokartwii
[2015-07-06 00:25:37 | GamespyBackendServer] mariokartwii servers: 1
[2015-07-06 00:25:38 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] RESPONSE: '\status\1\sesskey\94919902\statstring\/SCM/1/SCN/0/VER/90\locstring\final\updatepro\sesskey\94919902\firstname\Wii:3640650471079990@KMCE\partnerid\11\final\'...
[2015-07-06 00:25:38 | GamespyDatabase] [adCWhLOB] STARTING: SELECT COUNT(*) FROM ip_banned WHERE ipaddr = 192.168.101.197
[2015-07-06 00:25:38 | GamespyDatabase] [adCWhLOB] DONE: Took 0.000231981277466 real time / 0.0 processor time
[2015-07-06 00:25:38 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] {'status': '1', '**cmd_val**': '1', 'statstring': '/SCM/1/SCN/0/VER/90', '**cmd**': 'status', 'sesskey': '94919902', 'locstring': ''}
[2015-07-06 00:25:38 | GamespyDatabase] [7XekYIWB] STARTING: SELECT \* FROM buddies WHERE buddyProfileId = 4 AND status = 0
[2015-07-06 00:25:38 | GamespyDatabase] [7XekYIWB] DONE: Took 0.000603914260864 real time / 0.0 processor time
[2015-07-06 00:25:38 | GamespyDatabase] [rF0aYByR] STARTING: SELECT \* FROM buddies WHERE buddyProfileId = 4 AND status = 1 AND notified = 0
[2015-07-06 00:25:38 | GamespyDatabase] [rF0aYByR] DONE: Took 0.000550031661987 real time / 0.0 processor time
[2015-07-06 00:25:38 | GamespyDatabase] [xQeHo89V] STARTING: SELECT \* FROM buddies WHERE userProfileId = 4 AND blocked = 0
[2015-07-06 00:25:38 | GamespyDatabase] [xQeHo89V] DONE: Took 0.000210046768188 real time / 0.0 processor time
[2015-07-06 00:25:38 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] {'updatepro': '', 'firstname': 'Wii:3640650471079990@KMCE', '**cmd_val**': '', '**cmd**': 'updatepro', 'sesskey': '94919902', 'partnerid': '11'}
[2015-07-06 00:25:38 | GamespyDatabase] [BHarBU7V] STARTING: UPDATE users SET "firstname" = Wii:3640650471079990@KMCE WHERE profileid = 4
[2015-07-06 00:25:38 | GamespyDatabase] [BHarBU7V] DONE: Took 0.000910043716431 real time / 0.01 processor time
[2015-07-06 00:25:41 | GameSpyPlayerSearchServer] SEARCH RESPONSE: \otherslist\sesskey\94919902\profileid\4\numopids\0\opids\0\namespaceid\16\gamename\mariokartwii\final\
{'gamename': 'mariokartwii', '**cmd_val**': '', 'opids': '0', 'numopids': '0', '**cmd**': 'otherslist', 'sesskey': '94919902', 'profileid': '4', 'namespaceid': '16', 'otherslist': ''}
[2015-07-06 00:25:41 | GamespyDatabase] [fEcZd35T] STARTING: SELECT \* FROM users WHERE profileid = 0
[2015-07-06 00:25:41 | GamespyDatabase] [fEcZd35T] DONE: Took 0.00478100776672 real time / 0.0 processor time
[2015-07-06 00:25:41 | GameSpyPlayerSearchServer] SENDING: \otherslist\o\0\uniquenick\oldone\final\
[2015-07-06 00:25:59 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Received keep alive from 192.168.101.197:61798...
[2015-07-06 00:26:19 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Received keep alive from 192.168.101.197:61798...
[2015-07-06 00:26:39 | GameSpyQRServer] [192.168.101.197:61798 129bafce] Received heartbeat from 192.168.101.197:61798... localip010.0.1.30localport61798natneg1gamenamemariokartwiipublicip-1062705723publicport61798numplayers0maxplayers0unknown4unknown0unknown90unknown1unknown0unknown0unknown0
[2015-07-06 00:26:39 | GamespyBackendServer] Deleted 1 mariokartwii servers where session = 312192974
[2015-07-06 00:26:39 | GamespyBackendServer] Added {'natneg': '1', '**session**': 312192974, 'localport': '61798', 'unknown': '0', 'gamename': 'mariokartwii', 'numplayers': '0', 'maxplayers': '0', 'publicip': '-1062705723', '**console**': 0, 'localip0': '10.0.1.30', 'publicport': '61798'} to the server list for mariokartwii
[2015-07-06 00:26:39 | GamespyBackendServer] mariokartwii servers: 1
[2015-07-06 00:26:47 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] Client disconnected
[2015-07-06 00:26:47 | GamespyDatabase] [5tKH6gZO] STARTING: SELECT \* FROM buddies WHERE userProfileId = 4 AND blocked = 0
[2015-07-06 00:26:47 | GamespyDatabase] [5tKH6gZO] DONE: Took 0.000396013259888 real time / 0.0 processor time
[2015-07-06 00:26:47 | GamespyDatabase] [c0kyPlBs] STARTING: DELETE FROM sessions WHERE profileid = 94919902
[2015-07-06 00:26:47 | GamespyDatabase] [c0kyPlBs] DONE: Took 0.000447034835815 real time / 0.0 processor time
[2015-07-06 00:26:47 | GameSpyProfileServer] [192.168.101.197:61964 | 4 | RMCJ] Deleted session 
[2015-07-06 00:27:40 | GamespyBackendServer] Deleted 1 mariokartwii servers where session = 312192974
[2015-07-06 00:27:40 | GameSpyQRServer] Keep alive check removed 192.168.101.197:61798 for game mariokartwii. Client hasn't responded in 61 seconds.

The bottom line is, no matter what I do, I can't get error 23916 to appear for the "no name" or "Player" mii (the server calls the Player mii "no name")

**Comment by kyle95wm** [comment]
I guess nobody knows why this isn't working?
